### PR TITLE
make go covermod configurable

### PR DESCRIPTION
**Pull request by wonderflow**
Signed-off-by: 天元 <jianbo.sjb@alibaba-inc.com>

`go test -race` need CGO_ENABLED=1.

Tested with my personal repo: https://github.com/wonderflow/test-build

1. add build as submodule with this updated PR
2. run make test

### CODE REVIEW

**Comment by wonderflow** [comment]
ping @negz @hasheddan @jbw976  , please kindly review, thanks!

**Review by hasheddan — APPROVED** [review]
LGTM

**Review by jbw976 — COMMENTED** [review]
Thanks for this contribution @wonderflow! Can you please update the PR description with some details about how this has been tested? Thank you!

**Comment by wonderflow** [comment]
@jbw976  Thanks , I've tested it.

File: makelib/golang.mk [hunk 1]
```diff
@@ -129,6 +129,8 @@ endif
 GO_COMMON_FLAGS = $(GO_BUILDFLAGS) -tags '$(GO_TAGS)'
 GO_STATIC_FLAGS = $(GO_COMMON_FLAGS) $(GO_PKG_STATIC_FLAGS) -installsuffix static  -ldflags '$(GO_LDFLAGS)'
 GO_GENERATE_FLAGS = $(GO_BUILDFLAGS) -tags 'generate $(GO_TAGS)'
+GO_COVER_MODE ?= count
```
Comment (jbw976) [inline makelib/golang.mk:132]: i would recommend putting this up near the top of this file, around line 31 that says `# Optional build flags passed to go tools`.  That will help people more easily discover the set of config options that they can set.

nit: consider calling `CGO_ENABLED` -> `GO_CGO_ENABLED` instead, for someone doing a search for `^GO_` can find it along with the other options that are named similarly.

File: makelib/golang.mk [hunk 2]
```diff
@@ -171,7 +173,7 @@ ifeq ($(GO_NOCOV),true)
 else
 	@mkdir -p $(GO_TEST_OUTPUT)
 	@CGO_ENABLED=0 $(GOHOST) test -i -cover $(GO_STATIC_FLAGS) $(GO_PACKAGES) || $(FAIL)
```
Comment (jbw976) [inline makelib/golang.mk:175]: is it odd that this `test` command will always run with `CGO_ENABLED=0`, but the `test` command on the next line will use the `CGO_ENABLED` variable?

File: makelib/golang.mk (hunk 1 above)
Comment (wonderflow) [reply makelib/golang.mk]: Thanks, I followed the advice.

File: makelib/golang.mk (hunk 2 above)
Comment (wonderflow) [reply makelib/golang.mk]: Sure, I could change this line. But I noticed the `@CGO_ENABLED=0` was hard coded lots of places, I don't want this PR make widely changes

**Comment by wonderflow** [comment]
@jbw976  All fixed and again tested in  https://github.com/wonderflow/test-build, Thanks!

**Review by jbw976 — APPROVED** [review]
looks good @wonderflow! thank you for incorporating the feedback and for driving this improvement! 💪

**Pull request merged**